src/js/components/TextInput/TextInput.js

```diff
@@ -352,7 +352,7 @@ class TextInput extends Component {
     delete rest.onSuggestionsClose;
     const { showDrop, isInputEmpty } = this.state;
     const showStyledPlaceholder =
-      placeholder && typeof placeholder !== 'string' && isInputEmpty;
+      placeholder && typeof placeholder !== 'string' && isInputEmpty && !value;
```

Comment:
I don't think we need `isInputEmpty` still.

Reply:
in an uncontrolled component case we need to use isInputEmpty because we wont have the value prop.
do you think that situation could happen?

Reply:
Ah, good point. Agreed.